packages/contracts-bedrock/test/libraries/SafeCall.t.sol

```diff
@@ -3,59 +3,63 @@ pragma solidity 0.8.15;
 
 // Testing utilities
 import { Test } from "forge-std/Test.sol";
+import { StdCheatsSafe } from "forge-std/StdCheats.sol";
 
 // Target contract
 import { SafeCall } from "src/libraries/SafeCall.sol";
 
 contract SafeCall_Test is Test {
-    /// @dev Tests that the `send` function succeeds.
-    function testFuzz_send_succeeds(address from, address to, uint256 gas, uint64 value) external {
-        vm.assume(from.balance == 0);
-        vm.assume(to.balance == 0);
-        // no precompiles (mainnet)
-        assumeNotPrecompile(to);
-        // don't call the vm
-        vm.assume(to != address(vm));
-        vm.assume(from != address(vm));
-        // don't call the console
-        vm.assume(to != address(0x000000000000000000636F6e736F6c652e6c6f67));
-        // don't call the create2 deployer
-        vm.assume(to != address(0x4e59b44847b379578588920cA78FbF26c0B4956C));
-        vm.assume(to != address(this));
+    /// @notice Helper function to deduplicate code. Makes all assumptions required for these tests.
+    function assumeNot(address _addr) internal {
+        vm.assume(_addr.balance == 0);
+        vm.assume(_addr != address(this));
+        assumeAddressIsNot(_addr, StdCheatsSafe.AddressType.ForgeAddress, StdCheatsSafe.AddressType.Precompile);
+    }
 
-        assertEq(from.balance, 0, "from balance is 0");
-        vm.deal(from, value);
-        assertEq(from.balance, value, "from balance not dealt");
+    /// @notice Internal helper function for `send` tests
+    function sendTest(address _from, address _to, uint256 _gas, uint64 _value) internal {
+        assumeNot(_from);
+        assumeNot(_to);
 
-        uint256[2] memory balancesBefore = [from.balance, to.balance];
+        assertEq(_from.balance, 0, "from balance is 0");
+        vm.deal(_from, _value);
+        assertEq(_from.balance, _value, "from balance not dealt");
 
-        vm.expectCall(to, value, bytes(""));
-        vm.prank(from);
-        bool success = SafeCall.send(to, gas, value);
+        uint256[2] memory balancesBefore = [_from.balance, _to.balance];
+
+        vm.expectCall(_to, _value, bytes(""));
+        vm.prank(_from);
+        bool success;
+        if (_value == 0) {
+            success = SafeCall.send({ _target: _to, _value: _value });
+        } else {
+            success = SafeCall.send({ _target: _to, _gas: _gas, _value: _value });
+        }
```

Comment:
What is the relationship between value and gas, i.e. why do we always forward all gas when value is zero?

Reply:
Whoops, good catch here, will fix this

Reply:
the relationship is whether `gas` is or isn't zero. if it is, call `send` with the gas argument, which forwards  `gas`. if it isn't, call `send` with no gas argument, which forwards all the gas up to that point.